stack deploy: handle external network when deploying

vdemeester opened the pull request:
Fixes #28835. There is two commits :

- If the network is marked as external, don't use the namespace on it. Otherwise, it's not found.
- It also validate that the existing networks are available `before` scheduling any services. This is more consistent with the way `docker-compose` does it and helps making sure we don't schedule an incomplete stack.

```
$ docker stack deploy --compose-file docker-compose.yml foo
Ignoring unsupported options: build

Ignoring deprecated options:

container_name: Setting the container name is not supported.

network "outside" is declared as external, but could not be found. You need to create the network before the stack is deployed (with overlay driver)

```

/cc @dnephin @aanand @thaJeztah @mavenugo 

:frog: 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

aanand commented:
LGTM (IANAM)

dnephin commented on cli/command/stack/deploy.go:
```diff
@@ -179,18 +183,20 @@ func getConfigFile(filename string) (*composetypes.ConfigFile, error) {
 
 	for internalName, network := range networks {
 		if network.External.Name != "" {
+			externalNetworks = append(externalNetworks, network.External.Name)
```
I didn't notice this before but this check `network.Externalk.Name != ""` is wrong. I think it should be `if network.External.External`.

vdemeester replied:
@dnephin ok, it bothered me a little, I'll update :angel:

dnephin reviewed (commented):
one minor thing, otherwise LGTM

vdemeester committed:
stack deploy: handle external network when deploying

If the network is marked as external, don't use the namespace on
it. Otherwise, it's not found.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

dnephin reviewed (approved):
LGTM

vieux commented:
LGTM
